plugins/fileservices/fileservices.cpp

```diff
@@ -2774,19 +2829,44 @@ FILESERVICES_API void  FILESERVICES_CALL fsMoveExternalFile(ICodeContext * ctx,c
     AuditMessage(ctx,"MoveExternalFile",frompath,topath);
 }
 
+FILESERVICES_API void  FILESERVICES_CALL fsMoveExternalFile(ICodeContext *ctx,const char *location,const char *frompath,const char *topath)
+{
+    implementMoveExternalFile(ctx,location,frompath,topath,nullptr);
+}
+
+FILESERVICES_API void  FILESERVICES_CALL fsMoveExternalFile_v2(ICodeContext *ctx,const char *location,
+    const char *frompath,const char *topath,const char *planename)
+{
+    implementMoveExternalFile(ctx,location,frompath,topath,planename);
+}
+
 FILESERVICES_API void  FILESERVICES_CALL fsDeleteExternalFile(ICodeContext * ctx,const char *location,const char *path)
 {
     SocketEndpoint ep(location);
     if (ep.isNull())
         throw MakeStringException(-1,"fsDeleteExternalFile: Cannot resolve location %s",location);
     CDfsLogicalFileName lfn;
     lfn.setExternal(location,path);
-    checkExternalFileRights(ctx,lfn,false,true);
+
+    StringBuffer s("DeleteExternalFile ('");
     RemoteFilename rfn;
     rfn.setPath(ep,path);
     Owned<IFile> file = createIFile(rfn);
+    if (!file->exists())
+    {
+        checkExternalFileRights(ctx,lfn.get(),false,true);
+        s.appendf("fsDeleteExternalFile: File %s not found.')",path);
+        WUmessage(ctx,SeverityInformation,nullptr,s.str());
+        return;
```

Comment:
not clear why this is coded differently to MoveExternalFile.
This is check file exists 1st, and then checking the full scope (including the supposed filename tail).

Afaics, it should be coded in the same way that moveExternalFile is, i.e. only check full scope if (f->exists() && (f->isDirectory()==fileBool::foundYes)).

In fact, why not use checkExternalFilePath (with plane=nullptr) ?

Reply:
*But* - this JIRA/PR is titled "Add plane parameter to STD.File.MoveExternalFile"

You should either:
1) not make changes to DeleteExternalFile in this PR, and make it a separate JIRA/PR, OR
2) Change title/comments, and add a plane parameter to DeleteExternalFile at same time.

Reply:
> @jakesmith in the existing commits, both DeleteExternalFile and CreateExternalDirectory are changed because the changes in the checkExternalFileRights(). I do not want the changes in the checkExternalFileRights() to break those 2 functions. The original intention is not adding a plane parameter in those functions in this PR. If you want, I can roll back the changes in DeleteExternalFile and CreateExternalDirectory. Please let me know.

@wangkx - Given the knock on impact change to MoveExternalFile has had, and the required changes to DeleteExternalFile and CreateExternalDirectory, and the desire to add plane to those anyway, I would make them all consistent and add a plane parameter to both of these in this PR (and retitle the JIRA and commit.